lib/session.ex

```diff
@@ -0,0 +1,195 @@
+defmodule Janus.Session do
+  @moduledoc """
+  Keeps information about session and sends keep-alive messages
+  if given connection's transport requires it.
+  """
+
+  @type t :: pid()
+  @type message_t :: map()
+  @type connection_t :: pid()
+  @type timeout_t :: non_neg_integer()
+  @type session_id_t :: non_neg_integer()
+  @type plugin_t :: String.t()
+
+  @type plugin_handle_id :: pos_integer
+
+  @default_timeout 5000
+
+  use GenServer
```

Comment:
https://github.com/christopheradams/elixir_style_guide#modules
I usually use this guideline. We don't have any official rules though. Feel free to read that and change code if you agree with it.